Happy Thoughts - Talo PR

govargas opened the pull request:
[Live site](https://happythoughtstalo.netlify.app)
Everything's working fine, all requirements and stretch goals accomplished (I think). 
Check readme.md for more info

govargas committed:
set up tailwind, adding font families

govargas committed:
fix fonts

govargas committed:
create Form.jsx

govargas committed:
hook form into App.jsx

govargas committed:
requirements check

govargas committed:
added useEffect hook

govargas committed:
refactor dummy cards into ThoughtCard component

govargas committed:
added localStorage so user only likes once per session and across reloads

govargas committed:
created addThought function in App.jsx

govargas committed:
further work on like handling

govargas committed:
updated App.jsx

govargas committed:
edit form function in Form.jsx

govargas committed:
added setError to handleChange in Form.jsx

govargas committed:
fix client-side validation

govargas committed:
added styling for error when submitting

govargas committed:
fixed Send button

govargas committed:
update handleLike in App.jsx

govargas committed:
defined handleClick inside ThoughtCard.jsx

govargas committed:
wrapped heart button and like-count span inside a small flex container

govargas committed:
added character count

govargas committed:
character counter turns red after 140

govargas committed:
added animation, new thought fades in gently

govargas committed:
added aria-labels for accessibility

govargas committed:
updated readme.md

govargas committed:
fix readme.md

christina-baldwin commented:
> Netlify: [https://happythoughtstalo.netlify.app/](url)
> 
> Everything's working fine, all requirements and stretch goals accomplished (I think). Check readme.md for more info

Quick note before I check the code but the Netlify link isn't working for me.

christina-baldwin commented on src/App.jsx:
To try and reduce the amount of code in App I think you can handle the like functionality in the thought cards component. I'm not sure if this is actually important its just something I would do.

govargas replied:
Good point! Will do.

christina-baldwin reviewed (commented):
Didn't really have much to comment on as the code looks good to me, the components are organised and even though i can't access the Netlify link I remember the presented project from Friday and I know it works well. Great that you used Day.js and Tailwind CSS! Good job!

govargas commented:
Thanks for your comments, Christina!

Just a quick note about the Netlify link — when I first opened the PR, the project had a different Netlify URL. I updated the site name afterwards and edited the original comment with the new link, but for some reason it’s not clickable anymore. Not sure why — but the correct address is:
👉 [https://happythoughtstalo.netlify.app/](url)

govargas commented:
Just a quick heads-up about the Netlify link - when I first opened the PR, the site had a different URL. I updated the name afterwards and edited the original comment with the new link, but GitHub seems to be caching the old one or breaking the link somehow. 😅

The correct and working URL is:
[Live site](https://happythoughtstalo.netlify.app)

If it doesn’t open directly from here, try copy-pasting it into a new tab — it should work just fine.

Thanks again for taking the time to review the code!

christina-baldwin commented:
> Just a quick heads-up about the Netlify link - when I first opened the PR, the site had a different URL. I updated the name afterwards and edited the original comment with the new link, but GitHub seems to be caching the old one or breaking the link somehow. 😅
> 
> The correct and working URL is: [Live site](https://happythoughtstalo.netlify.app)
> 
> If it doesn’t open directly from here, try copy-pasting it into a new tab — it should work just fine.
> 
> Thanks again for taking the time to review the code!

That works! Thanks! And it looks great!

JennieDalgren commented on src/App.jsx:
```diff
@@ -1,5 +1,80 @@
+      );
+      // Record this ID in localStorage to prevent further likes
+      setLikedIds((prev) => {
+        const next = [...prev, id];
+        localStorage.setItem('happy-likes', JSON.stringify(next));
+        return next;
```
⭐

JennieDalgren commented on src/components/Form.jsx:
```diff
@@ -0,0 +1,121 @@
+        value={message}
+        onChange={handleChange}
+        disabled={submitting}
+        maxLength={140}
```
you can also set `minLength` here. 😎

JennieDalgren commented on src/components/Form.jsx:
```diff
@@ -0,0 +1,121 @@
+          ring-1 ring-stone-300
+          focus:outline-none focus:ring-2 focus:ring-pink-200
+        "
+        aria-invalid={error ? "true" : "false"}
+        aria-describedby={error ? "error-message" : undefined}
```
⭐

JennieDalgren commented on src/components/Form.jsx:
```diff
@@ -0,0 +1,121 @@
+          tracking-wide
+        "
+      >
+        {submitting ? 'Sending…' : '❤️ Send Happy Thought ❤️'}
```
small little detail that makes a huge difference! nice

JennieDalgren reviewed (dismissed):
Great job with the project, you have a tiny stylng bug to fix before I can approve this project
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/e63bca4e-7eea-4a09-9011-25729a92db46" />

Your code is clean, easy to follow and with the right amount of comments.
I agree with Christina that I would extract some of the logic into the components instead of playing it safe and keeping it in App.jsx

Nice that you implemented local storage as well as some nice styling touches.

Keep up the good work! 
and fix the overflowing text issue so this project can be approved.

govargas committed:
add Tailwind's word-wrapping utilities to prevent text overflow

govargas committed:
add minLength in Form.jsx

govargas committed:
reduce code in App.jsx by moving the like functionality to the ThoughtCard.jsx instead

govargas commented:
> Great job with the project, you have a tiny stylng bug to fix before I can approve this project <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/18549785/446298065-e63bca4e-7eea-4a09-9011-25729a92db46.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgxODk2NjcsIm5iZiI6MTc0ODE4OTM2NywicGF0aCI6Ii8xODU0OTc4NS80NDYyOTgwNjUtZTYzYmNhNGUtN2VlYS00YTA5LTkwMTEtMjU3MjlhOTJkYjQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTI1VDE2MDkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiNzNlM2ZkNTIzNjJjNDk4OTExNDEwYmQxYWNiM2VmN2QyYzg5ZmQ1NGMwYTA3OTU2ZjMyOTRiNDA2ZmY4N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UWf9jFMvlR9H6S6cgIRmFHqUCLLcMl4jy5e8htInqlc">
> 
> Your code is clean, easy to follow and with the right amount of comments. I agree with Christina that I would extract some of the logic into the components instead of playing it safe and keeping it in App.jsx
> 
> Nice that you implemented local storage as well as some nice styling touches.
> 
> Keep up the good work! and fix the overflowing text issue so this project can be approved.

Thanks for your feedback, Jennie. I fixed the overflow by using Tailwind's word-wrapping utilities and moved the like functionality from the App.jsx to the ThoughtCard.jsx instead. Hope it works as it should now!

HIPPIEKICK commented:
Please update to the new API link: https://happy-thoughts-api-4ful.onrender.com/

govargas committed:
update app with new API link

govargas commented:
> Please update to the new API link: https://happy-thoughts-api-4ful.onrender.com/

Hej Matilda! I just updated it, now it seems to work fine.

govargas committed:
moved js-project-api as a backend folder to this happy thoughts repo

govargas committed:
lock down the API so it only responds to the deployed frontend

govargas committed:
merge backend into monorepo complete

govargas committed:
move render url to a .env

govargas committed:
add paginated GET /thoughts with page, limit & meta

govargas committed:
feat: add edit/delete + schema validation for thoughts

govargas committed:
pagination up and running

govargas committed:
filter by likes: add a small input in the UI so users can show only thoughts with min X likes

govargas committed:
implement sort order toggle to let users choose to sort by newest first (default) or most-liked first

govargas committed:
implement infinite scrolling

govargas committed:
install bcrypt, jsonwebtoken and express-validator

govargas committed:
add User.js to backend/models

govargas committed:
add auth routes in backend/server.js

govargas committed:
backend fix, mount auth/thought routes

govargas committed:
auth: add register & login pages with react-router

govargas committed:
add a _redirects file for auth tests

govargas committed:
add netlify.toml for auth tests

govargas committed:
small fixes on register.jsx

govargas committed:
add global navigation & React Router routes for Home, Login, Register

govargas committed:
fix Register.jsx email bug

govargas committed:
fix loading error in App.jsx

govargas committed:
on thoughtRoutes.js: protect write routes & add atomic like endpoint

govargas committed:
auth+frontend: send JWT on all write requests (create, edit, delete, like)

govargas committed:
form: include Bearer token on POST /thoughts for authenticated submissions

govargas committed:
create Liked.jsx as a My Likes page and wire up route in App.jsx

govargas committed:
wire My Likes page with props for thoughts and handlers

govargas committed:
fix liked route

govargas committed:
fix(thought-card): unpack API response for likes, edits & deletes

govargas committed:
update readme.md with backend sprint info

govargas committed:
fix: ThoughtCard uses backend’s response wrapper for like/delete/update

govargas committed:
fetch likedIds from /auth/me on login and seed localStorage

govargas committed:
trying to fix 500 error when liking thoughts

govargas committed:
edit Thought.js and thoughtRoutes.js to try to fix 500 error

govargas committed:
edit thoughtRoutes.js so the like handler uses likedBy everywhere

govargas committed:
fix thoughtRoutes with an atomic update on the POST id like handler

Kasssiiii commented on src/pages/Register.jsx:
```diff
@@ -0,0 +1,88 @@
+          onChange={handleChange}
+          className="w-full border p-2"
+        />
+        {fieldErrors.email && <p className="text-red-500 text-sm">{fieldErrors.email}</p>}
```
It seems that i cannot log in and also don't receive an error message (used username kkk, password kkk and email a@b.c

Seems like backend doesn't support those new features

Kasssiiii commented on README.md:
```diff
@@ -1 +1,110 @@
+☑️ Clean code: Component-based structure, readable JSX, and DRY utility classes
+
+**Stretch Goals**  
+☑️ Live character counter under the textarea, turns red at 140+  
```
Couldn't type more than 140 characters (since it says 140+ characters). Could be an extra nice feature if extra characters turn red too.

Kasssiiii commented on backend/data/seed.js:
```diff
@@ -0,0 +1,31 @@
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://127.0.0.1:27017/happythoughts'
+
+// Build an absolute path to your JSON file
```
not sure if this is needed, you perhaps could load json file content w/ an import command

JennieDalgren replied:
This is correct, you can seed a json file in a simpler way, check the recordings or video material from the mongodb week

Kasssiiii commented:
Comment re: likes, delete, edit and add posts when not loggen in - seems like you can do it all while not logged in and the app doesn't send any warnings. Maybe that is on purpose. Perhaps it would be easier to disable all those functionalities when not logged it to save work on error handling

Kasssiiii commented on backend.zip:
not sure what is the purpose of this file :)

Kasssiiii commented on backend/middleware/auth.js:
```diff
@@ -0,0 +1,34 @@
+import User from '../models/User.js'
+
+// Ensure we have a secret configured
+if (!process.env.JWT_SECRET) {
```
nice! great feature that makes sure that everything is configures properly!!!

govargas committed:
adds message to tell the user
 to wait for the server to load

govargas committed:
bug fixes

govargas committed:
delete unnecesary folder

govargas committed:
clean up some comments

Kasssiiii commented on src/pages/Register.jsx:
```diff
@@ -0,0 +1,88 @@
+    const res = await fetch(`${API_URL}/auth/register`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
```
I tried to register on this endpoint on the live side and got this response. Doesn't seem this one supports login
<img width="1272" height="878" alt="Zrzut ekranu 2025-07-14 164308" src="https://github.com/user-attachments/assets/e049fcf5-c85c-4f02-8d1e-28eecfe1dc87" />

Kasssiiii commented on backend/models/User.js:
```diff
@@ -0,0 +1,40 @@
+
+// Virtual setter: when you assign user.password, it hashes into passwordHash
+userSchema.virtual('password')
+  .set(function(password) {
```
Nice solution with a virtual field. I learned something new, thank you!

Kasssiiii commented on backend/server.js:
```diff
@@ -0,0 +1,64 @@
+
+// Mount auth and thoughts routers
+app.use('/auth', userRoutes);
+app.use('/thoughts', thoughtRoutes);
```
Cool way of separating routes from different things

Kasssiiii commented on src/App.jsx:
```diff
@@ -1,5 +1,244 @@
+import Register from './pages/Register.jsx'
+import Liked from './pages/Liked.jsx'
+
+const API_URL = import.meta.env.VITE_API_URL
```
This seems like a lot of code for main App.jsx. I wonder if it would be better to divide parts of it into separate components

Kasssiiii commented on src/App.jsx:
```diff
@@ -1,5 +1,244 @@
+  // 🔹 2) Infinite scroll
+  useEffect(() => {
+    const handleScroll = () => {
+      if (loading || page >= meta.totalPages) return
```
Nice feature! Didn't expect to support infiniscroll would be that simple

Kasssiiii commented on src/components/Form.jsx:
```diff
@@ -0,0 +1,95 @@
+
+const API_URL = import.meta.env.VITE_API_URL
+
+export default function Form({ onSubmitThought }) {
```
Perheaps the name of the component could be more descriptive with re: to its function

Kasssiiii commented on src/components/Form.jsx:
```diff
@@ -0,0 +1,95 @@
+          {error}
+        </p>
+      )}
+      <button
```
Perhaps it should be more clearly indicated that one needs to be logged in in order to send the post

Kasssiiii commented on src/pages/Login.jsx:
```diff
@@ -0,0 +1,60 @@
+    if (!res.ok) {
+      setError(json.error || 'Login failed')
+    } else {
+      localStorage.setItem('token', json.token)
```
cool seeing local storage being used :)

Kasssiiii reviewed (commented):
Great job on the app, Talo! 
I had issues logging in (perhaps the app is hooked up to the wrong end-point). Also perhaps front-end and back-end parts of the project could be separated. 
Very nice features with infiniscroll and overal very nice feature set and super cute design. 
Great job!!!!

JennieDalgren commented on backend/models/Thought.js:
```diff
@@ -0,0 +1,34 @@
+  },
+  // Track exactly which users have liked this thought
+  likedBy: {
+    type: [{
+      type: Schema.Types.ObjectId,
+      ref: 'User'
+    }],
+    default: []
```
nice implementation

JennieDalgren reviewed (dismissed):
Great job with the project!

I can't register as a new user:
<img width="1387" height="805" alt="image" src="https://github.com/user-attachments/assets/7544ffb5-4dd8-468b-b63e-8f09363a2a1c" />

Please take a look at this when you have time, then I can continue the review

govargas committed:
Fix: Add missing register and login endpoints

Implemented POST /auth/register and /auth/login endpoints that were missing
from userRoutes.js, fixing 404 errors on user registration and login.

govargas committed:
Refactor: Use direct JSON import in seed file

Simplified seed.js by importing thoughts.json directly instead of using
fs.readFileSync(), removing unnecessary dependencies on fs and path modules.

govargas committed:
Fix: Correct JWT token field name in auth middleware

govargas commented:
Hi Jennie! I think I have fixed this issues now!
